File: backend/models/dog.py
from sqlalchemy import Column, Integer, String, JSON
from database import Base


class Dog(Base):
    __tablename__ = "dogs"

    id = Column(Integer, primary_key=True, index=True)
    external_id = Column(String, nullable=True, unique=True, index=True)  # e.g. "rg_12345"
    name = Column(String, nullable=False)
    age = Column(Integer)  # years
    breed = Column(String)
    size = Column(String)  # small, medium, large
    energy_level = Column(String)  # low, moderate, high
    temperament = Column(String)
    shelter = Column(String)
    city = Column(String)  # dc, nyc
    home_type_required = Column(String)  # apartment, house, any
    monthly_cost = Column(Integer)
    first_vet_days = Column(Integer)
    training_plan = Column(String)
    risk_flags = Column(JSON, default=list)
    behavior_notes = Column(String)
    image_url = Column(String, nullable=True)
    photos = Column(JSON, default=list)  # up to 6 photo URLs
    sex = Column(String, nullable=True)           # Male / Female
    weight_lbs = Column(Integer, nullable=True)  # extracted from description
    good_with = Column(String, nullable=True)    # e.g. "dogs, kids"
    neutered = Column(String, nullable=True)     # Yes / No / Unknown
    vaccinated = Column(String, nullable=True)   # Yes / No / Unknown
    house_trained = Column(String, nullable=True) # Yes / No / Unknown

File: ios/GoHomeNow/Models/Dog.swift
import Foundation

struct Dog: Codable, Identifiable {
    let id: Int
    let name: String
    let age: Int
    let breed: String
    let size: String
    let energyLevel: String
    let temperament: String
    let shelter: String
    let city: String
    let monthlyCost: Int
    let firstVetDays: Int
    let trainingPlan: String
    let riskFlags: [String]
    let behaviorNotes: String
    let imageUrl: String?
    var photos: [String]
    let sex: String?
    let weightLbs: Int?
    let goodWith: String?
    let neutered: String?
    let vaccinated: String?
    let houseTrained: String?
    var fitScore: Int?

    enum CodingKeys: String, CodingKey {
        case id, name, age, breed, size, temperament, shelter, city, photos, sex, neutered, vaccinated
        case energyLevel = "energy_level"
        case monthlyCost = "monthly_cost"
        case firstVetDays = "first_vet_days"
        case trainingPlan = "training_plan"
        case riskFlags = "risk_flags"
        case behaviorNotes = "behavior_notes"
        case imageUrl = "image_url"
        case weightLbs = "weight_lbs"
        case goodWith = "good_with"
        case houseTrained = "house_trained"
        case fitScore = "fit_score"
    }

    var fitScoreDisplay: String {
        guard let score = fitScore else { return "--" }
        return "\(score)%"
    }

    var fitScoreColor: String {
        guard let score = fitScore else { return "gray" }
        if score >= 85 { return "green" }
        if score >= 70 { return "orange" }
        return "red"
    }
}
